torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -23,28 +23,34 @@ XlaOpVector SizeNode::Lower(LoweringContext* loctx) const {
 }
 
 int64_t SizeNode::getStaticValue() const {
-  return operand(0).node->shape(0).size(dim_);
+  // Not all IR has torch::lazy::shape now, use xla::shape to unblock
+  // the development.
+  return dynamic_cast<const XlaNode*>(operand(0).node)
+      ->xla_shape(operand(0).index)
+      .dimensions(dim_);
 }
 
 bool SizeNode::isSymbolic() const {
-  auto symbolic_vec = operand(0).node->shape(0).is_symbolic();
-  if (!symbolic_vec.has_value()) {
-    return true;
-  }
-  return symbolic_vec->at(dim_);
+  // Not all IR has torch::lazy::shape now, use xla::shape to unblock
+  // the development.
+  return dynamic_cast<const XlaNode*>(operand(0).node)
```

Comment:
@Krovatkin @miladm  Something is off here. Let me explain my understanding.

from pytorch symint perspective
1. `isSymbolic == true` symint wraps around a Node
2. `isSymbolic == false` symint wraps around a int

This pr aims to provide `isSymbolic` using xla_shape, what I did was
1. `isSymbolic == true` this dimension is dynamic
2. `isSymbolic == false` this dimension is static

Noted that `Symbolic` != `Dynamic`, for example we can call `xla::GetDimensionSize(static_tensor, 0)` and it will return a Node represents size of `static_tensor` but `static_tensor` is not dynamic.

I guess the question is more of why do we ever need `isSymbolic` in the SizeNode. According to the defination, if we ever trigger the SizeNode computation, it will always be Symbolic. If none of the input is `Symbolic`, the regular python math operator will be used anyway.

Reply:
chatted with @Krovatkin , we agree that `isSymbolic` should be removed from SizeNode since it should always return true. `isSymbolic` should also equal to `isDynamic` for most real context. I will make it always return true in this pr and remove it once upstream remove it.